keras/src/quantizers/gptq.py

```diff
@@ -0,0 +1,350 @@
+from keras.src import ops
+from keras.src.layers import Dense
+from keras.src.layers import EinsumDense
+from keras.src.quantizers.gptq_quant import dequantize
+
+
+class GPTQ:
+    def __init__(self, layer):
+        self.original_layer = layer
+        self.num_samples = 0
+        self.quantizer = None
+
+        # Explicitly handle each supported layer type
+        if isinstance(layer, Dense) or (
+            isinstance(layer, EinsumDense) and layer.kernel.ndim == 2
+        ):
+            # For a standard Dense layer, the dimensions are straightforward.
+            self.kernel_shape = layer.kernel.shape
+            self.rows = self.kernel_shape[0]  # Input features
+            self.columns = self.kernel_shape[1]  # Output features
+            self.layer = layer  # The layer itself can be used directly.
+
+        # Handle 3D EinsumDense layers (typically from attention blocks).
+        elif isinstance(layer, EinsumDense) and layer.kernel.ndim == 3:
+            # For EinsumDense, we determine the effective 2D dimensions.
+            self.kernel_shape = layer.kernel.shape
+            shape = list(self.kernel_shape)
+            try:
+                d_model_dim_index = shape.index(max(shape))
+            except ValueError:
+                raise TypeError(
+                    f"Could not determine hidden dimension from shape {shape}"
```

Comment:
Can we add a little more context to this error message? Something like "Could not determine hidden dimension from shape {shape}. Expected kernel shape to be of the form ...."